### PR TITLE
[Merged by Bors] - Adding transform example links to documentation

### PR DESCRIPTION
**Pull request by bwhitt7**
# Objective

Working on issue #1934 , with linking examples to the documentation. PR for transform examples.

## Solution

Added to the documentation in bevy_transform transform.rs and global_transform.rs utilizing links from examples.

[X] 3d_rotations.rs linked to rotate in Transform
[X] global_vs_local_translation.rs linked to top of Transform and GlobalTransform documentation
[X] scale.rs linked to scale Struct in Transform
[X] transform.rs linked to top of Transform documentation
[X] translation.rs linked to from_translation in Transform

### CODE REVIEW

File: crates/bevy_transform/src/components/transform.rs [hunk 1]
```diff
@@ -48,6 +52,8 @@ impl Transform {
     /// Creates a new [`Transform`] at the position `(x, y, z)`. In 2d, the `z` component
     /// is used for z-ordering elements: higher `z`-value will be in front of lower
     /// `z`-value.
+    /// 
+    /// Example: [translations](https://github.com/bevyengine/bevy/blob/latest/examples/transforms/translation.rs)
```
Comment (alice-i-cecile) [inline crates/bevy_transform/src/components/transform.rs:56]: I think that this link would be a better fit on the field.

File: crates/bevy_transform/src/components/global_transform.rs [hunk 2]
```diff
@@ -25,6 +25,8 @@ use bevy_reflect::Reflect;
 /// This system runs in stage [`CoreStage::PostUpdate`](crate::CoreStage::PostUpdate). If you
 /// update the [`Transform`] of an entity in this stage or after, you will notice a 1 frame lag
 /// before the [`GlobalTransform`] is updated.
+/// 
+/// Example: [transform](https://github.com/bevyengine/bevy/blob/latest/examples/transforms/transform.rs),
```
Comment (alice-i-cecile) [inline crates/bevy_transform/src/components/global_transform.rs:29]: I think the global vs local transform example is important to call out here

**Comment by alice-i-cecile** [comment]
You can also see in the CI that it's complaining about lack of formatting :) https://github.com/bevyengine/bevy/actions/runs/3070501643/jobs/4960290437

That's an easy fix; just follow the directions there.

**Comment by alice-i-cecile** [comment]
Can you mention that this is for the Transform examples in your PR title? They get turned into commit names so it's useful to see what exactly was done at a glance.

File: crates/bevy_transform/src/components/transform.rs (hunk 1 above)
Comment (bwhitt7) [reply crates/bevy_transform/src/components/transform.rs]: gotcha! changed the location to the translate field

File: crates/bevy_transform/src/components/global_transform.rs (hunk 2 above)
Comment (bwhitt7) [reply crates/bevy_transform/src/components/global_transform.rs]: whoops! meant to put that there, changed it.

**Review by alice-i-cecile — APPROVED** [review]
Got there :D I'm happy with this now; I think that this will help improve discoverability nicely.

**Comment by alice-i-cecile** [comment]
@bwhitt7 you can press "Resolve conversation" on the comments you've addressed. It's very useful for reviewers so they can have a better idea of the current state of the PR.

**Comment by bwhitt7** [comment]
@alice-i-cecile gotcha! thank you for your help

File: crates/bevy_transform/src/components/transform.rs [hunk 3]
```diff
@@ -26,14 +26,22 @@ use std::ops::Mul;
 /// This system runs in stage [`CoreStage::PostUpdate`](crate::CoreStage::PostUpdate). If you
 /// update the [`Transform`] of an entity in this stage or after, you will notice a 1 frame lag
 /// before the [`GlobalTransform`] is updated.
+///
+/// Examples: [`transform`](https://github.com/bevyengine/bevy/blob/latest/examples/transforms/transform.rs), [`global_vs_local_translation.rs`](https://github.com/bevyengine/bevy/blob/latest/examples/transforms/global_vs_local_translation.rs)
```
Comment (djeedai) [inline crates/bevy_transform/src/components/transform.rs:30]: We should settle on either including or excluding the `.rs` extension, but avoid mixing styles.

File: crates/bevy_transform/src/components/transform.rs [hunk 4]
```diff
@@ -48,6 +56,7 @@ impl Transform {
     /// Creates a new [`Transform`] at the position `(x, y, z)`. In 2d, the `z` component
     /// is used for z-ordering elements: higher `z`-value will be in front of lower
     /// `z`-value.
+    ///
```
Comment (djeedai) [inline crates/bevy_transform/src/components/transform.rs:59]: Revert?

**Review by djeedai — CHANGES_REQUESTED** [review]
Looks mostly good, thanks for helping the documentation effort!

**Review by Nilirad — CHANGES_REQUESTED** [review]
Good job. I suggested to list the example links into a dedicated “Examples” section. Vertical space is not an issue for those items. The only exception was for the fields, where it's nice to have visually compact info.

I also converted link style to reference, for readability. Finally I removed the `.rs` extension in the link text.

File: crates/bevy_transform/src/components/transform.rs (hunk 4 above)
Comment (Nilirad) [reply crates/bevy_transform/src/components/transform.rs:59]:
```suggestion
```

**Comment by bwhitt7** [comment]
@Nilirad thank you for your suggestions, i think these are really nice changes. i've never used github much before, so I'm unsure where to go from here. would i commit your suggestions?

**Comment by alice-i-cecile** [comment]
You can go the Files Changed tab, add the suggestions to a batch and then commit the suggestions :)

**Comment by Nilirad** [comment]
You can also check this page to see how to apply suggestions:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

**Review by Nilirad — APPROVED** [review]
I checked the output. Looks OK overall.

**Comment by alice-i-cecile** [comment]
Congrats on your first merged PR :)

bors r+

**Pull request closed without merging**